ansible/roles/set-repositories/README.md

```diff
@@ -12,6 +12,36 @@ FIXME
 
 `use_content_view` - FIXME
 
+### Repo Method 'rhn' Variables
+
+**IMPORTANT:** When using `repo_method: rhn` only define the variables of one of the below options.
+
+**Option 1:** Register with credentials
+This option registers a system with Red Hat credentials and a pool id.
+If using this method define these variables in a secrets file.
+
+`rhel_subscription_user` (Required)
+User name of the Red Hat account that will register the system.
+This is the same user name used to log into access.redhat.com
+
+`rhel_subscription_pass` (Required)
+The password for the user name referenced in `rhel_subscription_user`.
+
+`rhsm_pool_ids` (Required)
+The pool id for the subscription to attach. By default `auto_attach` is false.
+
+**Option 2:** Register with an activation key.
+These variables will register a system with an activation key.
+
+`rhel_subscription_activation_key` (Required)
+The unique activation key created within an account at access.redhat.com
+
+`rhel_subscription_org_id` (Required)
+The organization ID of the the account which created the activation key.
+
+`rhsm_pool_ids` (Required)
+The pool id for the subscription to attach. By default `auto_attach` is false.
+
```

Comment:
This is a good write-up.

Can you translate it into ansible assertions ?

1. create `pre_checks_rhm.yml` file that would be included only if `repo_method` = `rhn`
2. use the `assert` module to ensure that the mandatory variables are set. Ensure also that variables mutually exclusive are not defined simultaneously, etc.

Here is an example using `assert`:

```yaml
- name: Ensure mandatory variables are set
  assert:
    that: "{{ check.that }}"
    fail_msg: "{{ check.msg }}"
    quiet: true
  loop_control:
    loop_var: check
    label: "{{ check.msg }}"
  loop:
    - msg: Variable guid is not defined
      that: gpte_reporting.guid is defined and gpte_reporting.guid != ''

    - msg: Variable env_type is not defined
      that: gpte_reporting.env_type is defined and gpte_reporting.env_type != ''
```